nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Run/UpdateHandlers/UpdateHandlerSelectionTests.cs

```diff
@@ -0,0 +1,183 @@
+    }
+
+    public static IEnumerable<object[]> GetUpdateHandlerFromJobTestData()
+    {
+        // to ensure we're not depending on any default values, _ALWAYS_ set the following properties explicitly:
+        //   Source
+        //   Dependencies
+        //   DependencyGroups
+        //   DependencyGroupToRefresh
+        //   SecurityUpdatesOnly
+        //   UpdatingAPullRequest
+
+        //
+        // group_update_all_versions
+        //
+        yield return
+        [
+            new Job()
+            {
+                Source = CreateJobSource("/"),
+                Dependencies = [],
+                DependencyGroups = [],
+                DependencyGroupToRefresh = null,
+                SecurityUpdatesOnly = false,
+                UpdatingAPullRequest = false,
+            },
+            GroupUpdateAllVersionsHandler.Instance,
+        ];
+
+        yield return
+        [
+            new Job()
+            {
+                Source = CreateJobSource("/"),
+                Dependencies = ["Dependency.A", "Dependency.B"],
+                DependencyGroups = [],
+                DependencyGroupToRefresh = null,
+                SecurityUpdatesOnly = true,
+                UpdatingAPullRequest = false,
+            },
+            GroupUpdateAllVersionsHandler.Instance,
+        ];
+
+        yield return
+        [
+            new Job()
+            {
+                Source = CreateJobSource("/"),
+                Dependencies = [],
+                DependencyGroups = [new() { Name = "some-group", AppliesTo = "security-updates" }],
+                DependencyGroupToRefresh = null,
+                SecurityUpdatesOnly = true,
+                UpdatingAPullRequest = false,
+            },
+            GroupUpdateAllVersionsHandler.Instance,
+        ];
+
+        //
+        // update_version_group_pr
+        //
+        yield return
+        [
+            new Job()
+            {
+                Source = CreateJobSource("/dir1", "/dir2"),
+                Dependencies = ["Some.Dependency"],
+                DependencyGroups = [new() { Name = "some-group", AppliesTo = "security-updates" }], // this
+                DependencyGroupToRefresh = "some-group",
+                SecurityUpdatesOnly = false,
+                UpdatingAPullRequest = true,
+            },
+            RefreshGroupUpdatePullRequestHandler.Instance,
+        ];
+
+        yield return
+        [
+            new Job()
+            {
+                Source = CreateJobSource("/src"),
+                Dependencies = ["Dependency.A", "Dependency.B"],
+                DependencyGroups = [new() { Name = "some-group", AppliesTo = "security-updates" }], // this
+                DependencyGroupToRefresh = "some-group",
+                SecurityUpdatesOnly = true,
+                UpdatingAPullRequest = true,
+            },
+            RefreshGroupUpdatePullRequestHandler.Instance,
+        ];
```

Comment:
An update job will also have an entry in `existing-pull-requests` or `existing-group-pull-requests` e.g. https://github.com/dependabot/smoke-tests/blob/5ee5f6adeb19861b022b2fdc5dcbfb01df58867d/tests/smoke-go-update-pr.yaml#L8-L10

This is important because there are cases where the Updater will close the PR being updated: https://github.com/dependabot/dependabot-core/blob/54ad7bd82aa21b0c11ad6f1ac39f993c4888b92c/updater/lib/dependabot/updater/operations/refresh_group_update_pull_request.rb#L102

Search the `updater` code for `close_pull_request` for the various reasons it would do that.

Reply:
In general I tried to closely copy the behavior of the Ruby code and I explicitly omitted the `dependency_group_empty` scenario because that's impossible due to this line that won't allow this update handler to be selected if there were no job dependencies.

https://github.com/dependabot/dependabot-core/blob/54ad7bd82aa21b0c11ad6f1ac39f993c4888b92c/updater/lib/dependabot/updater/operations/refresh_group_update_pull_request.rb#L36